### PR TITLE
Strip the version prefix before calling semver

### LINKED ISSUE #7052: run `minikube start` with --image-mirror-country=cn failed
When i use the **minikube 1.8.2** to start , it failed, But the 1.8.1 can work.

**The exact command to reproduce the issue**:

```
minikube start --cpus=4 --memory=4g --download-only=true --driver=kvm2 --image-mirror-country=cn --alsologtostderr -v=7
```
**The full output of the command that failed**:<details>
```

panic: semver: Parse(v1.17.3): Invalid character(s) found in major number "v1"

goroutine 1 [running]:
github.com/blang/semver.MustParse(0xc0007ee036, 0x7, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/pkg/mod/github.com/blang/semver@v3.5.0+incompatible/semver.go:319 +0x1c1
k8s.io/minikube/cmd/minikube/cmd.generateCfgFromFlags(0x2a45e60, 0xc0007ee036, 0x7, 0x1a6b53a, 0x4, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/app/cmd/minikube/cmd/start.go:825 +0x335
k8s.io/minikube/cmd/minikube/cmd.runStart(0x2a45e60, 0xc000148a80, 0x0, 0x8)
	/app/cmd/minikube/cmd/start.go:315 +0x3be
github.com/spf13/cobra.(*Command).execute(0x2a45e60, 0xc000148900, 0x8, 0x8, 0x2a45e60, 0xc000148900)
	/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:830 +0x2aa
github.com/spf13/cobra.(*Command).ExecuteC(0x2a44f60, 0x0, 0x1, 0xc000637bc0)
	/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914 +0x2fb
github.com/spf13/cobra.(*Command).Execute(...)
	/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
k8s.io/minikube/cmd/minikube/cmd.Execute()
	/app/cmd/minikube/cmd/root.go:108 +0x6a4
main.main()
	/app/cmd/minikube/main.go:66 +0xea

```

</details>

**The output of the `minikube logs` command**:<details>

</details>

**The operating system version**:

Ubuntu 18.04.4 LTS

### PR DESCRIPTION
**Pull request by afbjorklund**
Needs a mirror country unit test, apparently.

Closes #7052

### CODE REVIEW

**Comment by afbjorklund** [comment]
Test seems to be working as intended, catches the previous error which was the idea:

```
=== RUN   TestMirrorCountry
=== RUN   TestMirrorCountry/image-repository_none,_image-mirror-country_none
=== RUN   TestMirrorCountry/image-repository_auto,_image-mirror-country_none
panic: semver: Parse(v1.17.3): Invalid character(s) found in major number "v1" [recovered]
	panic: semver: Parse(v1.17.3): Invalid character(s) found in major number "v1"

```

It doesn't really _verify_ the repository though, since it could still be defined as empty:

```go
        // ImageRepositories contains all known image repositories
        ImageRepositories = map[string][]string{
                "global": {""},
                "cn":     {"registry.cn-hangzhou.aliyuncs.com/google_containers"},
        }

```

File: cmd/minikube/cmd/start_test.go [hunk 1]
```diff
@@ -70,6 +70,50 @@ func TestGetKuberneterVersion(t *testing.T) {
...
+			description:     "image-repository auto, image-mirror-country china",
+			imageRepository: "auto",
+			mirrorCountry:   "cn",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.description, func(t *testing.T) {
+			cmd := &cobra.Command{}
+			viper.SetDefault(imageRepository, test.imageRepository)
```
Comment (medyagh) [inline cmd/minikube/cmd/start_test.go]: Calling viper.Set in unit test has un-wanted side effects, such as setting the user 's settings and also can not run in parallel safely

Could we break down generateCfgFromFlags func so we can test only the mirror func without need to pass cobra ?

Comment (afbjorklund) [reply cmd/minikube/cmd/start_test.go]: No idea, think I only copied what the test above was doing or something

```
                        viper.SetDefault(kubernetesVersion, test.paramVersion)
```

Comment (medyagh) [reply cmd/minikube/cmd/start_test.go]: I think we can refactor our tests not to use viper as much as possible, we could parameterize the funcs into small independent testable units. what do u think? maybe good opprotunity to refactor this part of the code?

**Pull request merged**